### PR TITLE
Add tags for email notifications

### PR DESCRIPTION
**Pull request by lukelee1124**
Add tags field for email notifications when set and visible to user

### CODE REVIEW

File: core/email_api.php [hunk 1]
```diff
@@ -1635,6 +1640,12 @@ function email_build_visible_bug_data( $p_user_id, $p_bug_id, $p_message_id ) {
 	$t_category_name = category_full_name( $t_row['category_id'], false );
 	$t_bug_data['email_category'] = $t_category_name;
 
+	if( count( $t_tag_rows = tag_bug_get_attached( $p_bug_id ) ) !== 0 && access_compare_level( $t_user_access_level, config_get( 'tag_view_threshold' ) ) ) {
+		foreach( $t_tag_rows as $t_tag ) {
```
Comment (vboctor) [inline core/email_api.php]: I would use implode() with ', ' (comma + space) as the separator instead of the for loop.

**Comment by vboctor** [comment]
The body of the commit message should include "Fixes #20857".

File: core/email_api.php [hunk 2]
```diff
@@ -1635,6 +1640,12 @@ function email_build_visible_bug_data( $p_user_id, $p_bug_id, $p_message_id ) {
 	$t_category_name = category_full_name( $t_row['category_id'], false );
 	$t_bug_data['email_category'] = $t_category_name;
 
+	if( count( $t_tag_rows = tag_bug_get_attached( $p_bug_id ) ) !== 0 && access_compare_level( $t_user_access_level, config_get( 'tag_view_threshold' ) ) ) {
```
Comment (dregad) [inline core/email_api.php]: Move the assignment out of the if statement

File: core/email_api.php (hunk 1 above)
Comment (dregad) [inline core/email_api.php]: Not sure this can be done, since he's calling tag_get_link() on each tag.

**Comment by vboctor** [comment]
Replace with #770 due to several issues in this implementation.

**Pull request closed without merging**